src/app/icd/ICDManager.cpp

```diff
@@ -168,10 +223,13 @@ void ICDManager::UpdateOperationState(OperationalState state)
         }
 #endif
 
+        // Going back to Idle, all check-in messages are sent
+        mICDSenderPool.ReleaseAll();
```

Comment:
There should be documentation around where the allocation from the pool happens that makes it clear how the lifetimes are managed here.